src/component/Todos.js

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect, useState } from "react";
+import Navbar from "./Navbar";
+import { AiOutlinePlusCircle } from "react-icons/ai";
+import TodoItem from "./TodoItem";
+import moment from "moment";
+
+const Todos = () => {
+  const [todoList, setTodoList] = useState([]);
+  const [isModal, setIsModal] = useState(false);
+  const [error, setError] = useState(false);
+  const [editTodoId, setEditTodoId] = useState(null);
+  const [currentTimeAndDate, setCurrentTimeAndDate] = useState(
+    moment().format("YYYY-MM-DDTHH:mm")
+  );
+  const [todoInput, setTodoInput] = useState({
+    todoTitle: "",
+    time: currentTimeAndDate,
+  });
+  const currentDateAndTime = moment().format("YYYY-MM-DDTHH:mm"); //this is use for input datetime minimum value
+  //this function check for todo complete or not.
+  const hanldeIsComplete = (id) => {
+    setTodoList(
+      todoList.map((todo) => {
+        if (todo.id === id) {
+          return { ...todo, isComplete: !todo.isComplete };
+        } else {
+          return todo;
+        }
+      })
+    );
+  };
+
+  const handleChange = (e) => {
+    setTodoInput({ ...todoInput, [e.target.name]: e.target.value });
+  };
+
+  const handleModal = () => {
+    setIsModal(true);
+  };
+
+  const closeModal = () => {
+    setIsModal(false);
+    setEditTodoId(null);
+    setTodoInput({
+      todoTitle: "",
+      time: currentTimeAndDate,
+    });
+  };
+
+  //This function handleEditAndAddTodo
+  const hanldeEditAndAddTodo = () => {
+    if (todoInput.todoTitle.trim() === "") {
+      setError(true);
+      return;
+    }
+    //find todo index base on todoID
+    const todoIndex = todoList.findIndex((todo) => todo.id === editTodoId);
```

Comment:
you should be adding condition for editTodiId is present or not then go for todoIndex @amitark7